acceptable/tests/test_mocks.py

```diff
@@ -0,0 +1,124 @@
+        mock = EndpointMock(
+            call_recorder,
+            "service",
+            "api",
+            ['GET'],
+            "http://example.com",
+            request_schema={"type": "string"},
+            response_schema={"type": "number"},
+            response_callback=lambda req: response_tuple,
+        )
+        request = requests.Request("GET", "http://example.com", json="hello").prepare()
+        assert_that(mock._callback(request), Equals(response_tuple))
+
+    def test_validation_failure_event_mock(self):
+        call_recorder = CallRecorder()
+        mock = EndpointMock(
+            call_recorder,
+            "service",
+            "api",
+            ['GET'],
+            "http://example.com",
+            request_schema={"type": "number"},
+            response_schema=None,
+            response_callback=lambda req: (200, {}, b""),
+        )
+        request = requests.Request("GET", "http://example.com", json="hello").prepare()
+        with ExpectedException(AssertionError):
+            mock._callback(request)
+
+
+class ServiceTests(testtools.TestCase):
+    def make_test_service(self):
+        endpoints = [
+            EndpointSpec(
+                "test_endpoint",
+                "test-endpoint",
+                ["GET", "POST"],
+                {"type": "number"},
+                {"type": "number"},
+            ),
+            EndpointSpec(
+                "no_validation",
+                "no-validation",
+                ["GET"],
+                None,
+                None,
+            )
+        ]
+        service_factory = ServiceFactory(
+            "test-service",
+            endpoints,
+        )
+        service = service_factory("http://example.com")
+        service.endpoints.test_endpoint.set_response(json=999)
+        return service
+
+    def test_simple_endpoint_cm(self):
+        service = self.make_test_service()
+        with service.endpoints.test_endpoint() as mock:
+            requests.get("http://example.com/test-endpoint", json=888)
+            assert_that(mock.get_calls(), HasLength(1))
+
+    def test_service_cm(self):
+        service = self.make_test_service()
+        with service() as mock:
+            requests.get("http://example.com/test-endpoint", json=888)
+            assert_that(mock.endpoints.test_endpoint.get_calls(), HasLength(1))
+
+    def test_validation_failure(self):
+        service = self.make_test_service()
+        with service() as mock:
+            with ExpectedException(AssertionError):
+                requests.get("http://example.com/test-endpoint", json="string")
+        assert_that(mock.endpoints.test_endpoint.get_calls(), HasLength(1))
+
+    def test_responses_manager_resets_responses_mock(self):
+        service = self.make_test_service()
+        with service() as service_mock:
+            with responses_mock_context() as responses_mock:
+                responses_mock.add('GET', 'http://exmaple.com/responses-test', b'test')
```

Comment:
Is "exmaple" a deliberate typo for the test? If so, maybe a comment or more obvious typo?

Reply:
nope just a typo